e2e/cypress/integration/messaging/local_date_time_spec.js

```diff
@@ -49,46 +52,82 @@ describe('Messaging', () => {
     });
 
     describe('MM-21342 Post time should render correct format and locale', () => {
-        const re12Hour = /(?=.*\d{2}:\d{2} [AMP]{2})(^[0-9:\sAMP]{8}$)/; // matches 12-hour format with exact length of 8
-        const re24Hour = /(?=.*\d{2}:\d{2})(^[0-9:]{5}$)/; // matches 24-hour format with exact length of 5
-
-        [
+        const testCases = [
             {
                 name: 'in English',
                 channelIntro: 'Beginning of Channel Test',
                 locale: 'en',
                 manualTimezone: 'UTC',
                 localTimes: [
-                    {postIndex: 0, match12hour: '4:30 AM', match24hour: '04:30'},
-                    {postIndex: 1, match12hour: '12:30 PM', match24hour: '12:30'},
-                    {postIndex: 2, match12hour: '8:30 PM', match24hour: '20:30'},
+                    {postIndex: 0, standard: '4:30 AM', military: '04:30'},
+                    {postIndex: 1, standard: '12:30 PM', military: '12:30'},
+                    {postIndex: 2, standard: '8:30 PM', military: '20:30'},
+                    {postIndex: 3, standard: '12:30 AM', military: '00:30'},
                 ]
             },
             {
-                name: 'in other locale like Spanish',
+                name: 'in Spanish',
                 channelIntro: 'Inicio de Channel Test',
                 locale: 'es',
                 manualTimezone: 'UTC',
                 localTimes: [
-                    {postIndex: 0, match12hour: /4:30 a\. m\./, match24hour: '4:30'},
-                    {postIndex: 1, match12hour: /12:30 p\. m\./, match24hour: '12:30'},
-                    {postIndex: 2, match12hour: /8:30 p\. m\./, match24hour: '20:30'},
+                    {postIndex: 0, standard: '4:30 a. m.', military: '4:30'},
+                    {postIndex: 1, standard: '12:30 p. m.', military: '12:30'},
+                    {postIndex: 2, standard: '8:30 p. m.', military: '20:30'},
+                    {postIndex: 3, standard: '12:30 a. m.', military: '0:30'},
                 ]
             },
             {
-                name: 'for react-intl unsupported timezone',
+                name: 'in react-intl unsupported timezone',
                 channelIntro: 'Inicio de Channel Test',
                 locale: 'es',
                 manualTimezone: 'NZ-CHAT',
                 localTimes: [
-                    {postIndex: 0, match12hour: re12Hour, match24hour: re24Hour},
-                    {postIndex: 1, match12hour: re12Hour, match24hour: re24Hour},
-                    {postIndex: 2, match12hour: re12Hour, match24hour: re24Hour},
+                    {postIndex: 0, standard: '06:15 PM', military: '18:15'},
```

Comment:
We'll see if this passes CI (@saturninoabril, can you help verify?), but it /seemed/ unnecessary to have a regular expression here given all the inputs are deterministic.

Reply:
As I remembered, it's due to DST period on that timezone that makes it inconsistent but I'm good as is.  Let's see how it goes in the next cycle.

Reply:
If it starts to fail again, we can file a ticket and then push it off for a few weeks until it stops reproducing 😛